swarm/api/http/server_test.go

```diff
@@ -18,21 +18,113 @@ package http_test
 
 import (
 	"bytes"
+	"crypto/rand"
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"os"
 	"strings"
 	"sync"
 	"testing"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/swarm/api"
 	swarm "github.com/ethereum/go-ethereum/swarm/api/client"
 	"github.com/ethereum/go-ethereum/swarm/storage"
 	"github.com/ethereum/go-ethereum/swarm/testutil"
 )
 
+func init() {
+	log.Root().SetHandler(log.CallerFileHandler(log.LvlFilterHandler(log.LvlTrace, log.StreamHandler(os.Stderr, log.TerminalFormat(true)))))
+}
+
+// \TODO if create -> get -> update -> get, the last get with return 1.1 because 1.2 retrieve is still pending
+func TestBzzResource(t *testing.T) {
+	srv := testutil.NewTestSwarmServer(t)
+	defer srv.Close()
+
+	// our mutable resource "name"
+	keybytes := make([]byte, common.HashLength)
+	copy(keybytes, []byte{42})
+	srv.Hasher.Reset()
+	srv.Hasher.Write([]byte(fmt.Sprintf("%x", keybytes)))
+	keybyteshash := fmt.Sprintf("%x", srv.Hasher.Sum(nil))
+
+	// data of update 1
+	databytes := make([]byte, 666)
+	_, err := rand.Read(databytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// creates resource and sets update 1
+	url := fmt.Sprintf("%s/bzz-resource:/%x/13", srv.URL, keybytes)
+	resp, err := http.Post(url, "application/octet-stream", bytes.NewReader(databytes))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("err %s", resp.Status)
+	}
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(b, []byte(keybyteshash)) {
+		t.Fatalf("resource update hash mismatch, expected '%s' got '%s'", keybyteshash, b)
+	}
+	resp.Body.Close()
```

Comment:
This should be deferred right after getting the response so that it is called even if there is an error.